packages/mdc-auto-init/types.ts

```diff
@@ -0,0 +1,55 @@
+/**
+ * @license
+ * Copyright 2019 Google Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+export type MDCAutoInitKey = (
```

Comment:
Does this make autoinit completely inextensible to third-parties using TS? i.e. will this cause `register` with anything besides these values to fail to compile?

(Not to mention this adds one more thing we need to remember to populate any time we add a new component...)

Reply:
`tsc` will yell at us if we forget to populate it when we add a new component.

But you're right that this would make it inextensible to third parties.

Should I remove `MDCAutoInitKey` and change the param to accept a `string` instead?

Reply:
I think we should do that, yes. I'm pretty sure I've heard of clients registering additional components with autoinit.

Reply:
Done